torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6583,9 +6584,12 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                        # def the_method(i0):
                        #     return torch.polygamma(i0, 1)
                        #            ~~~~~~~~~~~~~~~ <--- HERE
-                       SkipInfo('TestCommon', 'test_variant_consistency_jit'),),
+                       SkipInfo('TestCommon', 'test_variant_consistency_jit'),
+                       SkipInfo('TestCommon', 'test_jit_alias_remapping'),
+                       SkipInfo('TestCommon', 'test_variant_consistency_eager')),
```

Comment:
This test fails because of the ordering of args in `polygamma`/`special_polygamma`: `(int, tensor)` but JIT tests expect first input to be a tensor.

```
======================================================================
ERROR: test_variant_consistency_eager_polygamma_polygamma_n_0_cpu_float32 (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
    result = test_fn(self, *args)
  File "/home/krshrimali/git/krshrimali/pytorch/torch/testing/_internal/common_device_type.py", line 266, in test_wrapper
    return test(*args, **kwargs)
  File "/home/krshrimali/git/krshrimali/pytorch/test/test_ops.py", line 345, in test_variant_consistency_eager
    _test_consistency_helper(samples, variants)
  File "/home/krshrimali/git/krshrimali/pytorch/test/test_ops.py", line 334, in _test_consistency_helper
    variant_forward = variant(cloned,
TypeError: special_polygamma(): argument 'n' (position 1) must be int, not Tensor
```

Reply:
This is not related to JIT but we use lambda to reorder `x, n` while calling polygamma
```python
op=lambda x, n, **kwargs: torch.polygamma(n, x, **kwargs)
```
This is because we only verify gradients in consistency_eager for input=Tensor (but not the ones in arg).

This test now fails because, we don't reorder the arguments for the alias. (thus it expects first argument to be int and not a Tensor)

To mitigate should we allow specifying lambda wrappers around aliases? Not sure but might need to investigate.

Relevant lines of test
* Here we acquire the actual operator of the mentioned alias.
https://github.com/pytorch/pytorch/blob/cf38b20c61e2b08496c51b0d879892f388d6e03b/test/test_ops.py#L288-L292

* Here we call the alias variant

https://github.com/pytorch/pytorch/blob/cf38b20c61e2b08496c51b0d879892f388d6e03b/test/test_ops.py#L354

Reply:
After an offline discussion with @krshrimali and thinking a bit more. We might as well just add a new entry for `special.polygamma` in OpInfo and maybe add a handcoded alias test for polygamma?

Reason:
* Even if we go through the hassle of making sure `test_variant_consistency_eager` works, we still don't get the `alias_remapping` test due to the use of lambda.
* I don't think there are other operators which might need this `op_aliases`. (It is particularly due to the peculiar signature of polygamma that we need this)

@mruberry what is your opinion? 
Thanks!

Reply:
I see the issue. Sure, implementing a new OpInfo for it (with a comment) sounds like a fine workaround.